config/externalname.go

```diff
@@ -109,6 +109,11 @@ var TerraformPluginFrameworkExternalNameConfigs = map[string]config.ExternalName
 	// OSIS Pipeline can be imported using the name
 	"aws_osis_pipeline": config.ParameterAsIdentifier("pipeline_name"),
 
+	// rds
+	//
+	// aws_rds_instance_state import format: rds_instance_state-instanceId-12345678
+	"aws_rds_instance_state": config.TemplatedStringAsIdentifier("", "rds_instance_state-{{ .parameters.identifier }}"),
```

Comment:
The [import](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_instance_state#import) section in the Terraform document seemed a bit contradictory to me. The document gives two different formats for importing this resource.
1. `db-L72FUFBZX2RRXT3HOJSIUQVOKE`
2. `rds_instance_state-instanceId-12345678`

The second format and the one we are using here seems a bit suspicious to me. We cannot observe the import test because the uptest is currently failing due to a connection error. Did you perform a manual import test?

Reply:
And from what I understand from the source [code](https://github.com/hashicorp/terraform-provider-aws/blob/459437a4bb7b20b49c797b07a1849a979cf6f7f7/internal/service/rds/instance_state.go#L151), the correct format seems to be the first one(db-...), can you please try the following configuration. And it would be great to verify this with an import test :) 
```suggestion
	"aws_rds_instance_state": config.TemplatedStringAsIdentifier("", "{{ .parameters.identifier }}"),
```

Reply:
@turkenf I updated the simplified configuration according to your suggestion.

Observations:
* Everything works in general according to the initial PR description. I still see `example-instancestate` 'internal' name as external name of instance state for some reason
```
k get -f examples/rds/v1beta1/instancestate.yaml
NAME                                                     SYNCED   READY   EXTERNAL-NAME           AGE
instancestate.rds.aws.upbound.io/example-instancestate   True     True    example-instancestate   15m

NAME                                                READY   SYNCED   EXTERNAL-NAME                   AGE
instance.rds.aws.upbound.io/example-instancestate   True    True     db-5XBUBIXX5VXQQTSMYIJTCTYRGQ   15m
```
* Import test. Import works even when `crossplane.io/external-name` is `whatever` . The resolved `forProvider.identifier` takes presedence and drives the import.
```
k get -f examples/rds/v1beta1/instancestate.yaml
NAME                                                     SYNCED   READY   EXTERNAL-NAME   AGE
instancestate.rds.aws.upbound.io/example-instancestate   True     True    whatever        5m7s

NAME                                                READY   SYNCED   EXTERNAL-NAME                   AGE
instance.rds.aws.upbound.io/example-instancestate   True    True     db-5XBUBIXX5VXQQTSMYIJTCTYRGQ   26m
```

So from the functional standpoint everything is working, the default extetnal-name propagation is slightly confusing